In [ ]:
import pandas as pd
pd.set_option('display.max_columns', None)

import numpy as np

import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats as stats

In [ ]:
df = pd.read_csv('marketing_campaign.csv', sep="\t")

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2240 entries, 0 to 2239
Data columns (total 29 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   ID                   2240 non-null   int64  
 1   Year_Birth           2240 non-null   int64  
 2   Education            2240 non-null   object 
 3   Marital_Status       2240 non-null   object 
 4   Income               2216 non-null   float64
 5   Kidhome              2240 non-null   int64  
 6   Teenhome             2240 non-null   int64  
 7   Dt_Customer          2240 non-null   object 
 8   Recency              2240 non-null   int64  
 9   MntWines             2240 non-null   int64  
 10  MntFruits            2240 non-null   int64  
 11  MntMeatProducts      2240 non-null   int64  
 12  MntFishProducts      2240 non-null   int64  
 13  MntSweetProducts     2240 non-null   int64  
 14  MntGoldProds         2240 non-null   int64  
 15  NumDealsPurchases    2240 non-null   i

People
- ID: Customer's unique identifier
- Year_Birth: Customer's birth year
- Education: Customer's education level
- Marital_Status: Customer's marital status
- Income: Customer's yearly household income
- Kidhome: Number of children in customer's household
- Teenhome: Number of teenagers in customer's household
- Dt_Customer: Date of customer's enrollment with the company
- Recency: Number of days since customer's last purchase
- Complain: 1 if the customer complained in the last 2 years, 0 otherwise

Products¶
- MntWines: Amount spent on wine in last 2 years
- MntFruits: Amount spent on fruits in last 2 years
- MntMeatProducts: Amount spent on meat in last 2 years
- MntFishProducts: Amount spent on fish in last 2 years
- MntSweetProducts: Amount spent on sweets in last 2 years
- MntGoldProds: Amount spent on gold in last 2 years

Promotion
- NumDealsPurchases: Number of purchases made with a discount
- AcceptedCmp1: 1 if customer accepted the offer in the 1st campaign, 0 otherwise
- AcceptedCmp2: 1 if customer accepted the offer in the 2nd campaign, 0 otherwise
- AcceptedCmp3: 1 if customer accepted the offer in the 3rd campaign, 0 otherwise
- AcceptedCmp4: 1 if customer accepted the offer in the 4th campaign, 0 otherwise
- AcceptedCmp5: 1 if customer accepted the offer in the 5th campaign, 0 otherwise
- Response: 1 if customer accepted the offer in the last campaign, 0 otherwise

Place
- NumWebPurchases: Number of purchases made through the company’s website
- NumCatalogPurchases: Number of purchases made using a catalogue
- NumStorePurchases: Number of purchases made directly in stores
- NumWebVisitsMonth: Number of visits to company’s website in the last month

In [ ]:
df.shape

(2240, 29)

In [ ]:
df.describe()

ID   Year_Birth         Income      Kidhome     Teenhome  \
count   2240.000000  2240.000000    2216.000000  2240.000000  2240.000000   
mean    5592.159821  1968.805804   52247.251354     0.444196     0.506250   
std     3246.662198    11.984069   25173.076661     0.538398     0.544538   
min        0.000000  1893.000000    1730.000000     0.000000     0.000000   
25%     2828.250000  1959.000000   35303.000000     0.000000     0.000000   
50%     5458.500000  1970.000000   51381.500000     0.000000     0.000000   
75%     8427.750000  1977.000000   68522.000000     1.000000     1.000000   
max    11191.000000  1996.000000  666666.000000     2.000000     2.000000   

           Recency     MntWines    MntFruits  MntMeatProducts  \
count  2240.000000  2240.000000  2240.000000      2240.000000   
mean     49.109375   303.935714    26.302232       166.950000   
std      28.962453   336.597393    39.773434       225.715373   
min       0.000000     0.000000     0.000000         0.000000   
25%      24.000000    23.750000     1.000000        16.000000   
50%      49.000000   173.500000     8.000000        67.000000   
75%      74.000000   504.250000    33.000000       232.000000   
max      99.000000  1493.000000   199.000000      1725.000000   

       MntFishProducts  MntSweetProducts  MntGoldProds  NumDealsPurchases  \
count      2240.000000       2240.000000   2240.000000        2240.000000   
mean         37.525446         27.062946     44.021875           2.325000   
std          54.628979         41.280498     52.167439           1.932238   
min           0.000000          0.000000      0.000000           0.000000   
25%           3.000000          1.000000      9.000000           1.000000   
50%          12.000000          8.000000     24.000000           2.000000   
75%          50.000000         33.000000     56.000000           3.000000   
max         259.000000        263.000000    362.000000          15.000000   

       NumWebPurchases  NumCatalogPurchases  NumStorePurchases  \
count      2240.000000          2240.000000        2240.000000   
mean          4.084821             2.662054           5.790179   
std           2.778714             2.923101           3.250958   
min           0.000000             0.000000           0.000000   
25%           2.000000             0.000000           3.000000   
50%           4.000000             2.000000           5.000000   
75%           6.000000             4.000000           8.000000   
max          27.000000            28.000000          13.000000   

       NumWebVisitsMonth  AcceptedCmp3  AcceptedCmp4  AcceptedCmp5  \
count        2240.000000   2240.000000   2240.000000   2240.000000   
mean            5.316518      0.072768      0.074554      0.072768   
std             2.426645      0.259813      0.262728      0.259813   
min             0.000000      0.000000      0.000000      0.000000   
25%             3.000000      0.000000      0.000000      0.000000   
50%             6.000000      0.000000      0.000000      0.000000   
75%             7.000000      0.000000      0.000000      0.000000   
max            20.000000      1.000000      1.000000      1.000000   

       AcceptedCmp1  AcceptedCmp2     Complain  Z_CostContact  Z_Revenue  \
count   2240.000000   2240.000000  2240.000000         2240.0     2240.0   
mean       0.064286      0.013393     0.009375            3.0       11.0   
std        0.245316      0.114976     0.096391            0.0        0.0   
min        0.000000      0.000000     0.000000            3.0       11.0   
25%        0.000000      0.000000     0.000000            3.0       11.0   
50%        0.000000      0.000000     0.000000            3.0       11.0   
75%        0.000000      0.000000     0.000000            3.0       11.0   
max        1.000000      1.000000     1.000000            3.0       11.0   

          Response  
count  2240.000000  
mean      0.149107  
std       0.356274  
min       0.000000  
25%       0.000000  
50%   

In [ ]:
df.head()

ID  Year_Birth   Education Marital_Status   Income  Kidhome  Teenhome  \
0  5524        1957  Graduation         Single  58138.0        0         0   
1  2174        1954  Graduation         Single  46344.0        1         1   
2  4141        1965  Graduation       Together  71613.0        0         0   
3  6182        1984  Graduation       Together  26646.0        1         0   
4  5324        1981         PhD        Married  58293.0        1         0   

  Dt_Customer  Recency  MntWines  MntFruits  MntMeatProducts  MntFishProducts  \
0  04-09-2012       58       635         88              546              172   
1  08-03-2014       38        11          1                6                2   
2  21-08-2013       26       426         49              127              111   
3  10-02-2014       26        11          4               20               10   
4  19-01-2014       94       173         43              118               46   

   MntSweetProducts  MntGoldProds  NumDealsPurchases  NumWebPurchases  \
0                88            88                  3                8   
1                 1             6                  2                1   
2                21            42                  1                8   
3                 3             5                  2                2   
4                27            15                  5                5   

   NumCatalogPurchases  NumStorePurchases  NumWebVisitsMonth  AcceptedCmp3  \
0                   10                  4                  7             0   
1                    1                  2                  5             0   
2                    2                 10                  4             0   
3                    0                  4                  6             0   
4                    3                  6                  5             0   

   AcceptedCmp4  AcceptedCmp5  AcceptedCmp1  AcceptedCmp2  Complain  \
0             0             0             0             0         0   
1             0             0             0             0         0   
2             0             0             0             0         0   
3             0             0             0             0         0   
4             0             0             0             0         0   

   Z_CostContact  Z_Revenue  Response  
0              3         11         1  
1              3         11         0  
2              3         11         0  
3              3         11         0  
4              3         11         0

## 1. Dari dataset yang diberikan:

- Hitung Probability Mass Function untuk jumlah kunjungan web bulanan (‘NumWebVisitsMonth’)



In [ ]:
df.shape

(2240, 29)

In [ ]:
selected_columns = df[['ID', 'NumWebVisitsMonth']]
selected_columns.head(20)

ID  NumWebVisitsMonth
0   5524                  7
1   2174                  5
2   4141                  4
3   6182                  6
4   5324                  5
5   7446                  6
6    965                  6
7   6177                  8
8   4855                  9
9   5899                 20
10  1994                  7
11   387                  8
12  2125                  2
13  8180                  6
14  2569                  8
15  2114                  3
16  9736                  8
17  4939                  7
18  6565                  5
19  2278                  6

In [ ]:
df['NumWebVisitsMonth'].unique()

array([ 7,  5,  4,  6,  8,  9, 20,  2,  3,  1, 10,  0, 14, 19, 17, 13])

In [ ]:
(df['NumWebVisitsMonth'] == 5).sum()

np.int64(281)

In [ ]:
x = 281
y = 2240
z = x / y
z

0.12544642857142857

In [ ]:
# Hitung frekuensi untuk setiap jumlah kunjungan
frequency = df['NumWebVisitsMonth'].value_counts().sort_index()

In [ ]:
# Hitung total data
total_data = len(df)

In [ ]:
# Hitung PMF (Probabilitas setiap jumlah kunjungan)
pmf = frequency / total_data

In [ ]:
pmf

NumWebVisitsMonth
0     0.004911
1     0.068304
2     0.090179
3     0.091518
4     0.097321
5     0.125446
6     0.151786
7     0.175446
8     0.152679
9     0.037054
10    0.001339
13    0.000446
14    0.000893
17    0.000446
19    0.000893
20    0.001339
Name: count, dtype: float64

- Berapa probabilitas seorang pelanggan memiliki 5 kunjungan web
bulanan? Bagaimana probabilitas ini dibandingkan dengan pelanggan
yang memiliki lebih dari 5 kunjungan?

In [ ]:
# Probabilitas seorang pelanggan memiliki 5 kunjungan web bulanan
prob_5_visits = pmf.get(5, 0)
print(f"\nProbabilitas seorang pelanggan memiliki 5 kunjungan web bulanan: {prob_5_visits}")

# Probabilitas pelanggan memiliki lebih dari 5 kunjungan web bulanan
prob_more_than_5_visits = pmf[pmf.index > 5].sum()
print(f"Probabilitas seorang pelanggan memiliki lebih dari 5 kunjungan web bulanan: {prob_more_than_5_visits}")


Probabilitas seorang pelanggan memiliki 5 kunjungan web bulanan: 0.12544642857142857
Probabilitas seorang pelanggan memiliki lebih dari 5 kunjungan web bulanan: 0.5223214285714287


In [ ]:
# Bandingkan probabilitas
if prob_5_visits > prob_more_than_5_visits:
    print("Peluang pelanggan dengan 5 kunjungan web lebih besar dibandingkan pelanggan dengan lebih dari 5 kunjungan.")
else:
    print("Peluang pelanggan dengan lebih dari 5 kunjungan web lebih besar dibandingkan pelanggan dengan 5 kunjungan.")

Peluang pelanggan dengan lebih dari 5 kunjungan web lebih besar dibandingkan pelanggan dengan 5 kunjungan.


2. Lakukan pengujian hipotesis untuk membuktikan apakah education
berpengaruh terhadap income atau tidak. Sebagai seorang DS cobalah untuk
menjelaskan Kesimpulan yang Anda dapatkan!

In [ ]:
df['Education'].unique()

array(['Graduation', 'PhD', 'Master', 'Basic', '2n Cycle'], dtype=object)

In [ ]:
edu_income = df[['Education', 'Income']]
edu_income.head(20)

Education   Income
0   Graduation  58138.0
1   Graduation  46344.0
2   Graduation  71613.0
3   Graduation  26646.0
4          PhD  58293.0
5       Master  62513.0
6   Graduation  55635.0
7          PhD  33454.0
8          PhD  30351.0
9          PhD   5648.0
10  Graduation      NaN
11       Basic   7500.0
12  Graduation  63033.0
13      Master  59354.0
14  Graduation  17323.0
15         PhD  82800.0
16  Graduation  41850.0
17  Graduation  37760.0
18      Master  76995.0
19    2n Cycle  33812.0

In [ ]:
# Hapus nilai yang hilang pada kolom 'Education' dan 'Income'
df_clean = df[['Education', 'Income']].dropna()

In [ ]:
# Hitung rata-rata pendapatan untuk setiap kategori pendidikan
education_groups_df = df_clean.groupby('Education')['Income'].mean().reset_index()
education_groups_df

Education        Income
0    2n Cycle  47633.190000
1       Basic  20306.259259
2  Graduation  52720.373656
3      Master  52917.534247
4         PhD  56145.313929

In [ ]:
# Buat grup data berdasarkan kategori pendidikan
grouped_data = [df_clean[df_clean['Education'] == education]['Income'] for education in df_clean['Education'].unique()]

# Lakukan uji ANOVA one way
anova_result = stats.f_oneway(*grouped_data)

# Cetak hasil ANOVA
print(f"F-statistic: {anova_result.statistic}")
print(f"P-value: {anova_result.pvalue}")

F-statistic: 27.73918235234821
P-value: 1.6677281316366318e-22


In [ ]:
# Kesimpulan berdasarkan nilai P-value
alpha = 0.05
if anova_result.pvalue <= alpha:
    print("Ada perbedaan signifikan dalam pendapatan berdasarkan tingkat pendidikan (Reject H₀)")
else:
    print("Tidak ada perbedaan signifikan dalam pendapatan berdasarkan tingkat pendidikan (Fail to reject H₀)")


Ada perbedaan signifikan dalam pendapatan berdasarkan tingkat pendidikan (Reject H₀)


### Hipotesis:

- **H₀ (Hipotesis Nol):** Tidak ada perbedaan signifikan dalam pendapatan antara kategori pendidikan yang berbeda. Ini berarti rata-rata pendapatan bagi semua kelompok pendidikan (Graduation, PhD, Master, Basic, 2n Cycle) adalah sama.
- **H₁ (Hipotesis Alternatif):** Ada perbedaan signifikan dalam pendapatan antara kategori pendidikan yang berbeda. Artinya, setidaknya satu kelompok pendidikan memiliki rata-rata pendapatan yang berbeda secara signifikan dari kelompok lain.

### Metode Pengujian:

Untuk menguji hipotesis ini, digunakan **ANOVA (Analysis of Variance)** one way. Metode ini cocok digunakan untuk membandingkan rata-rata pendapatan dari beberapa kelompok (kategori pendidikan dalam hal ini) untuk melihat apakah ada perbedaan yang signifikan.

### P-value:

- **P-value** yang dihasilkan dari uji ANOVA adalah **1.67e-22**, yang sangat kecil dan jauh di bawah tingkat signifikansi standar (α = 0.05).
- P-value ini menunjukkan probabilitas bahwa hasil yang kita amati (atau yang lebih ekstrem) terjadi secara kebetulan jika hipotesis nol benar.

### Significance Value (α):

- **Tingkat signifikansi (α)** yang digunakan adalah 0.05 (5%).
- Ini berarti bahwa jika p-value lebih kecil dari 0.05, kita akan menolak hipotesis nol (H₀) dan menyimpulkan bahwa ada perbedaan signifikan dalam pendapatan berdasarkan kategori pendidikan.

### Kesimpulan:

- Berdasarkan hasil uji ANOVA, p-value yang sangat kecil **(1.67e-22)** berarti kita **menolak hipotesis nol (H₀)**.
- Ini menunjukkan bahwa **ada perbedaan signifikan dalam pendapatan berdasarkan kategori pendidikan**.
  
### Interpretasi Rata-Rata Pendapatan:

Dari perhitungan rata-rata pendapatan per tingkat pendidikan:
- **Graduation**: 52720.37
- **PhD**: 56145.31
- **Master**: 52917.53
- **Basic**: 20306.26
- **2n Cycle**: 47633.19

Tingkat pendidikan yang lebih tinggi seperti **PhD** memiliki pendapatan rata-rata tertinggi, sedangkan **Basic** memiliki pendapatan rata-rata yang jauh lebih rendah dibandingkan dengan tingkat pendidikan lainnya. Hal ini menunjukkan bahwa pendidikan berperan penting dalam menentukan tingkat pendapatan.

Secara keseluruhan, hasil ini membuktikan bahwa pendidikan mempengaruhi pendapatan secara signifikan, dengan kelompok pendidikan yang berbeda menunjukkan perbedaan nyata dalam pendapatan rata-rata.